### PR TITLE
add tags and path sorting, add related docs

### LINKED ISSUE #1000: Sort API methods in OpenAPI (Swagger-UI)
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I sort these methods?  (POST > PUT > GET ...)

### Additional context
POST is written before PUT in code. 
image
image

### PR DESCRIPTION
**Pull request by Cajuteq**
Hi, having the need to sort the swagger documentation, I made a quick changes that resolves : https://github.com/tiangolo/fastapi/issues/1000
It relies on https://github.com/swagger-api/swagger-ui/issues/2990 for the underlying sort operation.
It might be a duplicate functionality regarding https://github.com/tiangolo/fastapi/pull/2337 but the way it works is way simpler as it relies heavily on swagger to do the work.
I also checked for Redoc ordering but reading https://github.com/Redocly/redoc/issues/98 it seems it is not done yet.
Please let me know if some changes are needed before you can merge, thanks for your work @tiangolo

### CODE REVIEW

File: docs/en/docs/advanced/extending-openapi.md [hunk 1]
```diff
@@ -248,3 +248,27 @@ Now, to be able to test that everything works, create a *path operation*:
 Now, you should be able to disconnect your WiFi, go to your docs at <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>, and reload the page.
 
 And even without Internet, you would be able to see the docs for your API and interact with it.
+
+
+### Add Tags
+
+You can group path by tags in both docs and redoc by using tags, tags also allows you to put descriptions and links. Tags description suppports markdown.
```
Comment (ArcLightSlavik) [inline docs/en/docs/advanced/extending-openapi.md:255]: We already have docs about tags here: https://fastapi.tiangolo.com/tutorial/path-operation-configuration/?h=tags#tags and here https://fastapi.tiangolo.com/tutorial/metadata/?h=tags#use-your-tags
So don't see a use for this?

File: docs/en/docs/advanced/extending-openapi.md [hunk 2]
```diff
@@ -248,3 +248,27 @@ Now, to be able to test that everything works, create a *path operation*:
...
+
+```Python hl_lines="9-22 56"
+{!../../../docs_src/extending_openapi/tutorial003.py!}
+```
+
+### Change the *path operation* order
+
+It is good to have a documentation, but a sorted one is better, you can use tags_sorter and operations_sorter to sort your docs.
+See this thread for the underlying logic (Swagger UI 3.17.6 required, Redoc not affected) :
+<a href="https://github.com/swagger-api/swagger-ui/issues/2990" class="external-link" target="_blank">https://github.com/swagger-api/swagger-ui/issues/2990</a>
```
Comment (ArcLightSlavik) [inline docs/en/docs/advanced/extending-openapi.md:265]: Is there a doc where all of them are listed? Dislike the fact it's linking to an issue, couldn't figure out the complete list myself.

File: docs/en/docs/advanced/extending-openapi.md [hunk 3]
```diff
@@ -248,3 +248,27 @@ Now, to be able to test that everything works, create a *path operation*:
...
+
+
+```Python hl_lines="37-38"
+{!../../../docs_src/extending_openapi/tutorial003.py!}
+```
+
+!!! tip
+    Possible values are "aplha" or a custom build *sort* function for the tags,
```
Comment (ArcLightSlavik) [inline docs/en/docs/advanced/extending-openapi.md:273]: A example method for "sort" would be 👍

**Review by ArcLightSlavik — COMMENTED** [review]
I do like this solution more than #2237 think realying heavily on swagger-ui is a good idea, on the other hand redoc not having this feature is a bit limiting so having it on our side could be nice? Personally never used redoc so this solution fits me more.

File: docs/en/docs/advanced/extending-openapi.md (hunk 1 above)
Comment (Cajuteq) [reply docs/en/docs/advanced/extending-openapi.md:255]: Well I must have missed the second one, and the first one uses a slightly different way of defining tags; 
Seeing this, I think the part about Tags sorting could be added to https://fastapi.tiangolo.com/tutorial/metadata/?h=tags#order-of-tags which is immediately after the second link you gave.

Also in the extending openapi doc we need to have tags defined so maybe we can just skip the Add Tags and instead put a link toward the doc here https://fastapi.tiangolo.com/tutorial/metadata/#metadata-for-tags

Let me know what you think so I can rewrite this part of the doc.

File: docs/en/docs/advanced/extending-openapi.md (hunk 3 above)
Comment (ArcLightSlavik) [reply docs/en/docs/advanced/extending-openapi.md:273]: Doesn't need to be *that* visual, just changing around 1 endpoint would be enough.

File: docs/en/docs/advanced/extending-openapi.md (hunk 2 above)
Comment (ArcLightSlavik) [reply docs/en/docs/advanced/extending-openapi.md:265]: Considering it's not in a separate table I doubt you could make a good highlight, think it's still better link than github issue, just need a way to point to the correct variables

File: docs/en/docs/advanced/extending-openapi.md (hunk 1 above)
Comment (ArcLightSlavik) [reply docs/en/docs/advanced/extending-openapi.md:255]: I like adding it to https://fastapi.tiangolo.com/tutorial/metadata/?h=tags#order-of-tags, the https://fastapi.tiangolo.com/tutorial/metadata/#metadata-for-tags seems like a different thing?

**Comment by Cajuteq** [comment]
So I put together the doc changes we talked about and made a small implementation tweak to allow custom function execution along with the custom function example. I am not quite happy with this version cause it force people to use '"alpha"' instead of "alpha" (mind the quotes). Maybe we could do a test on whether it's alpha/method and put quotes in those cases; although this would imply updating tests if SwaggerUI adds more ordering.

File: docs/en/docs/advanced/extending-openapi.md (hunk 1 above)
Comment (Cajuteq) [reply docs/en/docs/advanced/extending-openapi.md]: Hi, I included in the doc (metadata/order of tags) a link to the extending-openapi doc. metadata for tags is a bit different as it is the customization of the tags headers with markdown description and url link

File: docs/en/docs/advanced/extending-openapi.md (hunk 2 above)
Comment (Cajuteq) [reply docs/en/docs/advanced/extending-openapi.md]: I included the link to this general swagger doc and put as a technical note the link to the issue describing the technical implementation

File: docs/en/docs/advanced/extending-openapi.md (hunk 3 above)
Comment (Cajuteq) [reply docs/en/docs/advanced/extending-openapi.md]: I wrote a function in a generic manner using the length as ordering as it is simple to understand, effectively changes the display order in a easy to check way and as a bonus it allows to show the way of getting each entry "name" which is the operation path.

```
     operations_sorter='function(e,t) {\
            if (e.get("path").length < t.get("path").length) {return -1;}\
            if (e.get("path").length > t.get("path").length) {return 1;}\
            return 0;\
            }',
```

**Comment by Cajuteq** [comment]
@tiangolo can I hope this to be merged anytime soon ? Or is there anything to change in the proposed code, or the proposed doc update ?

File: docs/en/docs/advanced/extending-openapi.md [hunk 4]
```diff
@@ -248,3 +248,24 @@ Now, to be able to test that everything works, create a *path operation*:
 Now, you should be able to disconnect your WiFi, go to your docs at <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>, and reload the page.
 
 And even without Internet, you would be able to see the docs for your API and interact with it.
+
+
+### Change the *path operation* order
+
+It is good to have a documentation, but a sorted one is better, you can use tags_sorter and operations_sorter to sort your docs.
+
+Possible values are "aplha" or a custom build *sort* function for the tags, and "alpha", "method" or a custom build *sort* function for the operations as defined in the <a href="https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/" class="external-link" target="_blank">Swagger UI doc</a>.
```
Comment (erez-marmor) [inline docs/en/docs/advanced/extending-openapi.md:257]: typo: `aplha` -> `alpha`

**Comment by Cajuteq** [comment]
Hi, this functionality has been implemented as part of [#2568 ](https://github.com/tiangolo/fastapi/pull/2568 ) under the *swagger_ui_parameters* so it is no longer needed, I'm closing to keep thing clean, thank you all

**Pull request closed without merging**